device.go

```diff
@@ -59,6 +59,7 @@ const (
 type Device interface {
 	attach(hypervisor) error
 	detach(hypervisor) error
+	getDeviceType() string
```

Comment:
Just `type() string` please. See https://golang.org/doc/effective_go.html#Getters

Reply:
Changed this to deviceType(). golang isnt happy with type()